lib/matplotlib/quiver.py

```diff
@@ -85,6 +85,13 @@
     height of the axes, respectively, when the the window is resized;
     for 'dots' or 'inches', resizing does not change the arrows.
 
+  *orientation*: ['absolute' | 'relative']
```

Comment:
Excellent question!  In fact, if one uses  `angles='xy'`, then the original behavior is as expected; this was actually the intention for the case of looking at a gradient.  The new `orientation='relative`' is presently *wrong* for this case--it defeats the formerly correct behavior.  So the original bug report was actually a failure to recognize that `angles='xy'` was needed for the OP's use case--that's what it was designed for.  If anything like the `orientation` kwarg is to be retained, this PR needs to be modified to exclude the `angles='xy'` case, with corresponding changes to the docstring.